src/whylogs/core/statistics/stringtracker.py

```diff
@@ -0,0 +1,245 @@
+import logging
+import math
+
+from datasketches import frequent_strings_sketch
+
+from whylogs.core.statistics.thetasketch import ThetaSketch
+from whylogs.core.summaryconverters import from_string_sketch
+from whylogs.proto import CharPosMessage, CharPosSummary, StringsMessage, StringsSummary
+from whylogs.util import dsketch
+from whylogs.util.util_functions import encode_to_integers
+
+from .numbertracker import NumberTracker
+
+MAX_ITEMS_SIZE = 128
+MAX_SUMMARY_ITEMS = 100
+
+logger = logging.getLogger(__name__)
+
+
+class CharPosTracker:
+    def __init__(self, character_list: str = "abcdefghijklmnopqrstuvwzyz0123456789-"):
+
+        self.character_list = character_list
+        self._character_list_map = {x: idx for idx, x in enumerate(character_list)}
```

Comment:
If we want to work out of the box we need need a default, since this only instantiated with the column is created, we can have a method there to define the character list which can be pasted with the track API. is that reasonable ?

Reply:
I think we're talking about different things here.
* Out of box character support: yes, we should have a predefined list, and ideally users can customize them
* Memory efficiency: just because a character is listed in the list, we should **eagerly** create a map entry for them. These entries can be lazily created (missing = empty values, otherwise) -> this is what I was trying to address in my comment.

Reply:
done